README.en

```diff
@@ -0,0 +1,357 @@
+const Scanner2&, Scanner1::State&, Scanner2::State&, const char* begin, const char* end)
+function. This function checks the string faster than two consecutive calls of
+the regular Run() function.
+
+Also, there is the RunHelper class that simplifies working with the Initialize-Step-Run-Step-Final
+cycle writing it as a single expression (see an example).
+
+Four more functions helpful in lexical analysis are LongestPrefix(), LongestSuffix(), ShortestPrefix(),
+and ShortestSuffix(). These functions returns the largest / the smallest prefix or suffix allowed by
+the pattern. Before LongestSuffix() or ShortestSuffix() call, it is required to run Fsm::Reverse().
+
+Machine union
+================
+
+As it is already mentioned above, the time to check whether the string matches the pattern
+does not depend on the pattern size. In terms of the checking speed it is better to use
+some complex expressions than a lot of simple ones. If it is required to check the text with
+a lot of simple expressions, it is possible to unite several scanners into one.
+
+The function Pire::Scanner::Glue() unites several machines. The resulting scanner
+checks whether the string matches all united patterns at a time. The Scanner::RegexpsCount()
+function returns common number of patterns. It is recommended to replace running Final() with
+AcceptedRegexps(const State&) that returns iterators with the [begin,end) semantics indicating the
+range of expression numbers to which the read string corresponds.
+
+Numbers start from 0 and, after union, numbers of expressions on the right are shifted
+by the expression numbers on the left.
+
+United machines can be united once again. However, it is not recommended to repeat union
+indefinitely because the number of states in the resulting scanner grows
+exponentially with each added pattern. The max size of the scanners can be specified in
+the maxSize parameter in the function Glue(). If the specified size is exceeded, an empty scanner
+will be returned (Size() == 0, Empty() == true).
+
+Parsing a regular expression
+============================
+
+An instruction above describes analyzing the text with the ready scanner and this
+block contains information on how to get the scanner. The easiest way to get the scanner
+is to construct it from the string with the regular expression using the Pire::Lexer class.
+
+Regular expressions have the syntax similar to POSIX. It supports standard features as
+a|b, a*, a+, ., [a-z], a{3}, a{3,5}, a{3,}) and character classes (\w (alphabetic characters),
+\W (non-alphabetic characters), \d (decimal characters), \D (non-digit character), \s (whitespace),
+\S (non-whitespace)). Operators a&b (intersection, see Fsm::operator &), ~a (inversion, see
+Fsm::Complement), non-greedy analogs of repetitions *?, +?, ?? (Fsm::SlowCapturingScanner).
+Operators in this scanner can be set to as short expressions as possible, in contrast to greedy
+analogs that set to as long expressions as possible.
```

Comment:
The source paragraph

> Кроме того, могут быть добавлены
> операции a&b (пересечение, см. Fsm::operator &) и ~a (инверсия, см. Fsm::Complement).
> Помимо этого, для Fsm::SlowCapturingScanner есть возможность использовать нежадные аналоги
> повторений - операторы *?, +?, ??. В этом сканере данные операторы будут принимать
> как можно более короткие выражения, в отличие от жадных аналогов (которые, наоборот, принимают
> как можно более длинные.

is hard for google translate.
But after tweaking both the source and its suggested translation I have got

> In addition, a&b (intersection, see Fsm :: operator &) and ~a (inverse, see Fsm :: Complement) can be used.
> Also, Fsm :: SlowCapturingScanner has the ability to use non-greedy repetition operators - * ?, + ?, ??.
> These operators will accept as short subtexts as possible, as opposed to greedy counterparts (which, on the contrary, accept as long as possible).

Our FSMs are [acceptors](https://en.wikipedia.org/wiki/Finite-state_machine#Acceptors). So the words "to accept" and "to match" can be used interchangeably. Whilst the word "to set" is not suitable here.